src/modules/timeslots/time-slots.service.ts

```diff
@@ -0,0 +1,72 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { ModuleRepository } from '../modules.repository';
+import { CreateTimeSlotDTO } from './dto/create.time-slots.dto';
+import { TimeSlotDTO } from './dto/time-slots.dto';
+import { TimeSlotEntity } from './time-slots.entity';
+import { TimeSlotRepository } from './time-slots.repository';
+
+@Injectable()
+export class TimeSlotsService {
+  constructor(private timeSlotRepo: TimeSlotRepository, private moduleRepo: ModuleRepository) {}
+
+  async create(moduleId: number, dto: CreateTimeSlotDTO): Promise<TimeSlotDTO> {
+    const module = await this.moduleRepo.getById(moduleId);
+    if (module == null) {
+      throw new NotFoundException();
+    }
+    const entity = await this.timeSlotRepo.saveOrUpdate(
+      new TimeSlotEntity({
+        ...dto,
+        module,
+      }),
+    );
+    console.log(entity.id);
```

Comment:
console.log should be in production code

Reply:
Oh 😅 didn't see that one